solr/contrib/ltr/src/java/org/apache/solr/ltr/feature/SolrFeature.java

```diff
@@ -237,21 +237,25 @@ public FeatureScorer scorer(LeafReaderContext context) throws IOException {
      * @return DocIdSetIterator to traverse documents that matched all filter
      *         criteria
      */
+    // TODO it's not optimal to call getProcessedFilter per-segment!  Save the results into one Query
+    // TODO rename to "FromFilterQueries" suffix to at least suggest this uses the filter cache
     private DocIdSetIterator getDocIdSetIteratorFromQueries(List<Query> queries,
         LeafReaderContext context) throws IOException {
       final SolrIndexSearcher.ProcessedFilter pf = ((SolrIndexSearcher) searcher)
           .getProcessedFilter(null, queries);
-      final Bits liveDocs = context.reader().getLiveDocs();
-
-      DocIdSetIterator idIter = null;
-      if (pf.filter != null) {
-        final DocIdSet idSet = pf.filter.getDocIdSet(context, liveDocs);
-        if (idSet != null) {
-          idIter = idSet.iterator();
-        }
+      if (pf.postFilter != null) {
+        throw new SolrException(SolrException.ErrorCode.BAD_REQUEST,
+            "PostFilter queries are not supported");
       }
-
-      return idIter;
+      Query q = pf.filter;
+      if (q == null) {
+        q = new MatchAllDocsQuery(); // usually never happens?
+      }
+      Scorer scorer = q.createWeight(searcher, ScoreMode.COMPLETE_NO_SCORES, 1f).scorer(context);
```

Comment:
Is it guaranteed that the query is rewritten at this stage?

Reply:
Ah; good catch!  No it's not.